### DISCUSSION TITLE
Region error when deploying to GOV

### DISCUSSION THREAD

**Discussion by lundejd** [microsoft/finops-toolkit#820] [General]
I'm trying to deploy to Azure GOV but when I run the template and get to the deploy step the validation fails with:

"{
  "code": "LocationNotAvailableForResourceType",
  "message": "The provided location 'eastus2' is not available for resource type 'Microsoft.ManagedIdentity/userAssignedIdentities'. List of available regions for the resource type is 'usgovvirginia,usgoviowa,usgovtexas,usgovarizona,usdodeast,usdodcentral'."
}"

I don't see "eastus2" in the template or variables but something seems to want to deploy there. Looking at my settings, I have all regions selected correctly.

![Screenshot 2024-07-13 at 8 58 55 AM](https://github.com/user-attachments/assets/6491fb60-552b-41a0-ba5b-67dd53540054)

**Comment by flanakin** [comment]
What version of hubs is this? Please note FinOps hubs 0.2+ don't work in Azure Gov today because Cost Management doesn't support FOCUS exports. You can add a mapping file for actual or amortized cost to get it to work, but this would require some extra setup on your part.

Assuming this is 0.4, we didn't add the Azure Gov supported regions for EventGrid since it's not fully supported. That said, we should do that anyway. I'll take another look at this and switch it to use fallback logic for the known unsupported regions and rely on the specified eventGridLocation for anything else. That should minimize the risk of this in the future.

**Comment by lundejd** [comment]
I clicked “Deploy to Azure Gov” below the note that says to use FinOps hubs 0.1.1 so I assumed that is what was being used, but maybe not. Wouldn’t expect there to be a button if it didn’t work… but I will look closer. Thanks.

From: Michael Flanakin ***@***.***>
Date: Monday, July 22, 2024 at 5:58 PM
To: microsoft/finops-toolkit ***@***.***>
Cc: John Lundell ***@***.***>, Author ***@***.***>
Subject: Re: [microsoft/finops-toolkit] Region error when deploying to GOV (Discussion #820)

What version of hubs is this? Please note FinOps hubs 0.2+ don't work in Azure Gov today because Cost Management doesn't support FOCUS exports. You can add a mapping file for actual or amortized cost to get it to work, but this would require some extra setup on your part.

Assuming this is 0.4, we didn't add the Azure Gov supported regions for EventGrid since it's not fully supported. That said, we should do that anyway. I'll take another look at this and switch it to use fallback logic for the known unsupported regions and rely on the specified eventGridLocation for anything else. That should minimize the risk of this in the future.
